plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/config/CPSConfiguration.java

```diff
@@ -24,33 +24,49 @@
 
 package org.jenkinsci.plugins.workflow.cps.config;
 
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
 import edu.umd.cs.findbugs.annotations.NonNull;
 import hudson.Extension;
 import hudson.ExtensionList;
 import jenkins.model.GlobalConfiguration;
 import jenkins.model.GlobalConfigurationCategory;
+import org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval;
 import org.jenkinsci.Symbol;
 
+/**
+ * @deprecated
+ * This class has been deprecated and its only configuration value is ignored. Do not rely on it or use it in any way.
+ * In order to force using the system sandbox for pipelines, please use the flag
+ * org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval.get().isForceSandbox
+ * or
+ * org.jenkinsci.plugins.scriptsecurity.scripts.ScriptApproval.get().isForceSandboxForCurrentUser
+ **/
 @Symbol("cps")
 @Extension
+@Deprecated
 public class CPSConfiguration extends GlobalConfiguration {
 
     /**
      * Whether to show the sandbox checkbox in jobs to users without Jenkins.ADMINISTER
      */
-    private boolean hideSandbox;
+    private transient boolean hideSandbox;
 
     public CPSConfiguration() {
-        load();
-    }
 
-    public boolean isHideSandbox() {
-        return hideSandbox;
-    }
+        load();
+        if (hideSandbox) {
+            ScriptApproval.get().setForceSandbox(hideSandbox);
+        }
 
-    public void setHideSandbox(boolean hideSandbox) {
```

Comment:
Did you try retaining the setter, and having it just delegate to `ScriptApproval.setForceSandbox` after printing a stack trace? I think that would retain compatibility for JCasC users as well.

Reply:
Restoring CASC compatibility